### PR TITLE
deno: 1.3.3 -> 1.4.0 -> 1.5.0

### PR DESCRIPTION
**Pull request by 06kellyjac**
###### Motivation for this change

Bump `deno` to ~`1.4.0`~ `1.5.0`
Removed test plugin lib and test server bin

Depends on #96890
~Will undraft and rebase once it's merged~ now in master

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS (x86_64)
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -9580,15 +9580,22 @@ in
...
   rust_1_45 = callPackage ../development/compilers/rust/1_45.nix {
     inherit (darwin.apple_sdk.frameworks) CoreFoundation Security;
+    llvmPackages = llvmPackages_10;
+  };
+  rust_1_46 = callPackage ../development/compilers/rust/1_46.nix {
+    inherit (darwin.apple_sdk.frameworks) CoreFoundation Security;
+    llvmPackages = llvmPackages_10;
   };
-  rust = rust_1_45;
+  rust = rust_1_46;
```
Comment (Mic92) [inline pkgs/top-level/all-packages.nix:9593]: Someone should check what version we use for building firefox right now and if this is affected by this change.

Comment (06kellyjac) [reply pkgs/top-level/all-packages.nix:9593]: :+1: 
ofborg might have done a build already but I'm not too familiar with where to look for that
https://github.com/NixOS/nixpkgs/pull/96890#issuecomment-684509529

I don't plan on merging this until `rust_1_46` is in master either as default or as a separate optional version.

#96890

**Comment by 06kellyjac** [comment]
rust `1.46` now merged. Going to rebase

**Comment by solson** [comment]
Deno has moved up to version 1.4.4.

**Comment by 06kellyjac** [comment]
Thanks, I'm aware. Just no real reason to keep bumping versions until rust `1.46` is in master.

If there are people who have a `1.46` overlay or something and want me to keep this PR up to date to actually use deno `1.4.x` before it's merged then I can.

**Comment by solson** [comment]
Ahh, that makes sense. I don't personally need that, I just misunderstood what state this PR was in.

**Comment by 06kellyjac** [comment]
No worries, thanks for taking interest.
I'm checking pretty much every day as to whether it's made it from staging to master but I doubt it'll move over until 20.09 is released.

**Comment by 06kellyjac** [comment]
Result of `nixpkgs-review pr 97975` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>deno</li>
  </ul>
</details>

**Comment by Mic92** [comment]
Result of `nixpkgs-review pr 97975` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>deno</li>
  </ul>
</details>

**Pull request merged**